api/v1/jobs/PKPJobController.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+/**
+ * @file api/v1/jobs/PKPJobController.php
+ *
+ * Copyright (c) 2014-2023 Simon Fraser University
+ * Copyright (c) 2003-2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPJobController
+ *
+ * @ingroup api_v1_jobs
+ *
+ * @brief Handle API requests for jobs
+ *
+ */
+
+ namespace PKP\API\v1\jobs;
+
+use Closure;
+use APP\facades\Repo;
+use PKP\security\Role;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use PKP\core\PKPBaseController;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Route;
+
+class PKPJobController extends PKPBaseController
+{
+    public function nonContextualApi(): bool
+    {
+        return true;
+    }
+
+    public function getHandlerPath(): string
+    {
+        return 'jobs';
+    }
+
+    public function getRouteGroupMiddlewares(): array
+    {
+        $roles = implode('|', [Role::ROLE_ID_SITE_ADMIN]);
+
+        return [
+            "has.user",
+            "has.roles:{$roles}",
+        ];
+    }
+
+    public function getGroupRoutesCallback(): Closure
+    {
+        return function() {
+            
+            Route::get('all', [static::class, 'getJobs'])
```

Comment:
Let's use first-class callables when possible: `$this->getJobs(...)` [should work](https://laravel-code.tips/you-can-use-first-class-callables-for-routes-in-php-81/), I think.

Reply:
Update . But note that it's only useable for now as we have decided keep the routes in their own controller classes . If we decide to move it to separate `routes` file, then we have to user something like `ControllerClass@method` or [Controller::class, 'method'] syntax .